source/custom/nidaqmx_service.custom.cpp

```diff
@@ -235,6 +236,7 @@ ::grpc::Status NiDAQmxService::ReadAnalogWaveforms(::grpc::ServerContext* contex
 
     response->set_samps_per_chan_read(samples_per_chan_read);
     response->set_status(status);
+    context->AddTrailingMetadata("ni-samps-per-chan-read", std::to_string(samples_per_chan_read));
```

Comment:
Move this above `return ConvertApiErrorStatusForTaskHandle`. The objective is to return the metadata along with the error.